assets/scripts/main.js

```diff
@@ -1,21 +1,29 @@
+//Upon Page load, call activateSidebarHandler()
 document.addEventListener('DOMContentLoaded', () => {
-    const buttons = document.querySelectorAll('button');
-    const editBox = document.getElementById('editBox');
-    const save = document.querySelector('.save');
-    const userInput = document.getElementById('userInput');
+    activateSidebarHandler();
+});
 
-    buttons.forEach((button) => {
-        // When a Button is clicked, editBox appears
-	button.addEventListener('click', () => {
-		selectedButton = button;
-		editBox.style.display = 'block';
+// If any button is clicked in the sidebar, call editFortuneBox(button), which takes an argument of the button that was clicked
+function activateSidebarHandler() {
+    const sidebarButtons = document.querySelectorAll('.sidebarButton');
+    
+    sidebarButtons.forEach((button) => {
+        button.addEventListener('click', () => {
+            editFortune(button);
+        });
+    });
+}
 
-        // When the save button is clicked, button's text is changed to the input from editBox
-		save.addEventListener('click', () => {
-			button.textContent = userInput.value;
-			editBox.style.display = 'none';
-		});
+// Make the editFortuneBox appear, allow the user to write in their own fortune, and change the text of the button that was passed in
+function editFortune(button){
+    const editFortuneBox = document.getElementById('editFortuneBox');
+    const saveButton = document.querySelector('.saveButton');
+    const userInput = document.getElementById('userInput');
 
-	 });
-     });
-});
+    editFortuneBox.style.display = 'block';
+
+    saveButton.addEventListener('click', () => {
+        button.textContent = userInput.value;
+        editFortuneBox.style.display = 'none';
+    });
```

Comment:
We don't replace `textContent` before or after the user gives a value, meaning every time they edit it will be prefilled with the last thing they submitted. Imo, it should have the fortune they are currently editing instead -- warrants discussion and an ADR @adarsh249